### PR TITLE
Implement new pull to load more function

### PR DESCRIPTION
**Pull request by villyGreen**

### CODE REVIEW

File: Example/Example/Examples/RequestScalingHeader.swift [hunk 1]
```diff
@@ -12,55 +12,80 @@ import ScalingHeaderScrollView
 struct RequestScalingHeader: View {
 
     @Environment(\.presentationMode) var presentationMode
-
```
Comment (f3dm76) [inline Example/Example/Examples/RequestScalingHeader.swift:15]: this separates @Environment (global) block of variables from private block, please return the \n

File: Example/Example/Examples/RequestScalingHeader.swift [hunk 2]
```diff
@@ -12,55 +12,80 @@ import ScalingHeaderScrollView
...
-    
+    @State private var colorSet: [Color] = [.red, .blue, .green, .black, .pink, .purple, .yellow, .red]
+
     var body: some View {
         ZStack(alignment: .topLeading) {
             ScalingHeaderScrollView {
                 ZStack {
                     Rectangle()
-                        .fill(Color.gray.opacity(0.15))
-                    Image(uiImage: image)
```
Comment (f3dm76) [inline Example/Example/Examples/RequestScalingHeader.swift:26]: why did you remove the header? the library is about the header which scales, and all of the examples need to show that - that's the main point. load more is just an extra feature. please restore the header

File: Source/ScalingHeaderScrollView.swift [hunk 3]
```diff
@@ -68,6 +80,9 @@ public struct ScalingHeaderScrollView<Header: View, Content: View>: View {
     /// Allow enlarging while pulling down
     private var allowsHeaderGrowthFlag: Bool = false
 
+    /// Use this variable to set the padding value of the content from the edge when pull-to-load is active
+    private var pullToLoadContentOffset: CGFloat = 0.0
```
Comment (f3dm76) [inline Source/ScalingHeaderScrollView.swift:84]: `pullToLoadMoreContentOffset`
you should always use the whole thing in every connected variable - `pullToLoadMore`: `pullToLoadMoreProgress` `shouldPullToLoadMore` etc. Don't throw away words to make it shorter, and don't rephrase to make it seemingly more fit to some exact use-case. Just use `pullToLoadMore` every time it has to do with this feature, then it will be much easier to maintain

**Review by f3dm76 — CHANGES_REQUESTED** [review]

File: Example/Example/Examples/RequestScalingHeader.swift [hunk 4]
```diff
@@ -12,55 +12,80 @@ import ScalingHeaderScrollView
 struct RequestScalingHeader: View {
 
     @Environment(\.presentationMode) var presentationMode
-    
     @State private var isLoading: Bool = false
-    @State private var image: UIImage = UIImage()
-    @ObservedObject private var imageLoader = ImageLoaderService()
-    
+    @State private var colorSet: [Color] = [.red, .blue, .green, .black, .pink, .purple, .yellow, .red]
```
[the remark this answers is not in the document]
Comment (villyGreen) [reply Example/Example/Examples/RequestScalingHeader.swift:16]: Because the shuffleDataSource function is called, which shuffles the order in the array, and in a comment below, you suggested adding 5 new elements to this array

[the remark this answers is not in the document]
Comment (f3dm76) [reply Example/Example/Examples/RequestScalingHeader.swift:17]: ah, got it. then better use each constant only once, like this
private let colorSet = [.red, .blue, .green, .black, .pink, .purple, .yellow]
@State private var displayedColors: [Color] = []

.onAppear {
displayedColors = (0..<5).map { _ in colorSet.randomElement() }
}

.pullToLoadMore(isLoading: $isLoading, contentOffset: 32) {
        DispatchQueue.main.asyncAfter(deadline: .now() + 3) {
            colorSet += (0..<5).map { _ in colorSet.randomElement() }
            isLoading = false
        }
    }

File: Example/Example/Examples/RequestScalingHeader.swift [hunk 5]
```diff
@@ -12,55 +12,86 @@ import ScalingHeaderScrollView
...
-                imageLoader.load()
+            .pullToLoadMore(isLoading: $isLoading, contentOffset: 32) {
+                DispatchQueue.main.asyncAfter(deadline: .now() + 3) {
+                    colorSet += [.red, .blue, .green, .yellow, .pink].shuffled()
+                    isLoading = false
+                }
             }
+            .height(min: 90.0, max: 250)
             .ignoresSafeArea()
-
```
Comment (f3dm76) [inline Example/Example/Examples/RequestScalingHeader.swift:45]: Please return \n
